File: src/models/entrada.java
package models;

import java.sql.Date;

public class entrada {

	public entrada(String asunto, String fecha,  String area, boolean confidencial, boolean urgente) {
		super();
		this.asunto = asunto;
		this.fecha = fecha;
		this.area = area;
		this.confidencial = confidencial;
		this.urgente = urgente;
		
		this.observaciones = "";
		//this.soloCoronel = false;
		this.tramitado = false;
		this.canalEntrada = "";
		
	}
	
	int id;
	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	public String getAsunto() {
		return asunto;
	}
	public void setAsunto(String asunto) {
		this.asunto = asunto;
	}
	public String getFecha() {
		return fecha;
	}
	public void setFecha(String fecha) {
		this.fecha = fecha;
	}
	public String getFile() {
		return file;
	}
	
	public String getArea() {
		return area;
	}
	
	public void setArea(String area) {
		this.area = area;
	}
	public void setFile(String file) {
		this.file = file;
	}

	
	
	public boolean isConfidencial() {
		return confidencial;
	}
	public void setConfidencial(boolean confidencial) {
		this.confidencial = confidencial;
	}
	public boolean isUrgente() {
		return urgente;
	}
	public void setUrgente(boolean urgente) {
		this.urgente = urgente;
	}
	public String getObservaciones() {
		return observaciones;
	}
	public void setObservaciones(String observaciones) {
		this.observaciones = observaciones;
	}
	
//	public boolean isSoloCoronel() {
//		return soloCoronel;
//	}
//	public void setSoloCoronel(boolean soloCoronel) {
//		this.soloCoronel = soloCoronel;
//	}
	public boolean isTramitado() {
		return tramitado;
	}
	public void setTramitado(boolean tramitado) {
		this.tramitado = tramitado;
	}
	
	String asunto;
	String fecha;
	String file;
	String area;
	String observaciones;
	String canalEntrada;
	String numEntrada;
	String tramitadoPor;

	public String getTramitadoPor() {
		return tramitadoPor;
	}
	public void setTramitadoPor(String tramitadoPor) {
		this.tramitadoPor = tramitadoPor;
	}
	public comentario getComentario() {
		return comentario;
	}
	public void setComentario(comentario comentario) {
		this.comentario = comentario;
	}
	public String getNumEntrada() {
		return numEntrada;
	}
	public void setNumEntrada(String numEntrada) {
		this.numEntrada = numEntrada;
	}

	boolean urgente;
	boolean confidencial;
	//boolean soloCoronel;
	boolean tramitado;
	public comentario comentario;

	

	
	
	public String getCanalEntrada() {
		return canalEntrada;
	}
	public void setCanalEntrada(String canalEntrada) {
		this.canalEntrada = canalEntrada;
	}
	
	
}

File: src/models/entrada.js
const Comentario = require('./comentario'); // Assuming Comentario class is in the same directory

class Entrada {
    constructor(asunto, fecha, area, confidencial, urgente) {
        this.id = null; // Or generate a unique ID
        this.asunto = asunto;
        this.fecha = fecha;
        this.file = null; // Initialize as null or appropriate default
        this.area = area;
        this.observaciones = "";
        this.canalEntrada = "";
        this.numEntrada = "";
        this.tramitadoPor = "";
        this.urgente = urgente;
        this.confidencial = confidencial;
        this.tramitado = false;
        this.comentario = new Comentario(this.id); // Create a Comentario object, pass entrada's id
    }

    // Getters
    getId() { return this.id; }
    getAsunto() { return this.asunto; }
    getFecha() { return this.fecha; }
    getFile() { return this.file; }
    getArea() { return this.area; }
    getObservaciones() { return this.observaciones; }
    getCanalEntrada() { return this.canalEntrada; }
    getNumEntrada() { return this.numEntrada; }
    getTramitadoPor() { return this.tramitadoPor; }
    isUrgente() { return this.urgente; }
    isConfidencial() { return this.confidencial; }
    isTramitado() { return this.tramitado; }
    getComentario() { return this.comentario; }

    // Setters
    setId(id) {
        this.id = id;
        if (this.comentario) {
            this.comentario.setEntradaId(id); // Keep Comentario's entrada_id in sync
        }
    }
    setAsunto(asunto) { this.asunto = asunto; }
    setFecha(fecha) { this.fecha = fecha; }
    setFile(file) { this.file = file; }
    setArea(area) { this.area = area; }
    setObservaciones(observaciones) { this.observaciones = observaciones; }
    setCanalEntrada(canalEntrada) { this.canalEntrada = canalEntrada; }
    setNumEntrada(numEntrada) { this.numEntrada = numEntrada; }
    setTramitadoPor(tramitadoPor) { this.tramitadoPor = tramitadoPor; }
    setUrgente(urgente) { this.urgente = urgente; }
    setConfidencial(confidencial) { this.confidencial = confidencial; }
    setTramitado(tramitado) { this.tramitado = tramitado; }
    setComentario(comentario) { this.comentario = comentario; }
}

module.exports = Entrada;
